core/container.go

```diff
@@ -1656,7 +1656,40 @@ func (container *Container) ImageRefOrErr(ctx context.Context) (string, error) {
 	return "", errors.Errorf("Image reference can only be retrieved immediately after the 'Container.From' call. Error in fetching imageRef as the container image is changed")
 }
 
-func (container *Container) AsService(ctx context.Context) (*Service, error) {
+type ContainerAsServiceArgs struct {
+	// Command to run instead of the container's default command
+	Args []string `default:"[]"`
+
+	// If the container has an entrypoint, prepend it to this exec's args
+	UseEntrypoint bool `default:"false"`
+
+	// Content to write to the command's standard input before closing
+	Stdin string `default:""`
+
+	// Redirect the command's standard output to a file in the container
+	RedirectStdout string `default:""`
+
+	// Redirect the command's standard error to a file in the container
+	RedirectStderr string `default:""`
+
+	// Exit codes this exec is allowed to exit with
+	Expect ReturnTypes `default:"SUCCESS"`
```

Comment:
:thinking: so are these args the duplicate of `WithExec`? If so, we should embed the `core.ContainerExecOpts` struct to avoid duplication and prevent drift.

But also, not entirely sure, since some of these aren't *strictly* the same right now. See https://github.com/dagger/dagger/blob/d50fee49415d9802add4d6c6588a842ed855e953/core/service.go#L434-L445

Not all of the ops from the `WithExec` are actually propagated through to a container:
- `Stdin`, `RedirectStdout`, `RedirectStderr` are actually misleading, because they're actually passed into `startContainer`. Even if you *could* redirect output, you can't read files from a running container (yet).
- `Expect` doesn't affect anything, since the service's exit code is really just discarded right?
- `NoInit` doesn't seem to do anything?

I think we should probably remove these options from the new `AsService` method entirely, better to not have them, than to have them there and be confusing.

Reply:
> 🤔 so are these args the duplicate of WithExec

One of the difference is that `Args` is optional in `AsService` api.

> NoInit doesn't seem to do anything?

hmm, this seems to be the case. But that means it didn't had any effect with the current `AsService` api either. I'll double check this.

> Expect doesn't affect anything, since the service's exit code is really just discarded right?

yeah, that makes sense. I'll double check once, and then remove this and other suggested fields from AsService api.

Reply:
> hmm, this seems to be the case. But that means it didn't had any effect with the current `AsService` api either. I'll double check this.

Yeah good point! I do think that having this on `AsService` directly though *feels* different than getting a previous `WithExec` to run as a service. If we're moving to a cleaner API, I think it's fair to clear this ambiguity up!